### PR TITLE
Remove mutable wrappers from props

### PR DESCRIPTION
**Pull request by nnelgxorz**
Remove the mutable wrapper functions from props. Everything seems to work the same as before, but this should allow for better treeshaking of data on the server.

### CODE REVIEW

**Comment by gioboa** [comment]
I added mutable because Qwik complains about them 🤔

**Comment by nnelgxorz** [comment]
There was an issue early on when props were made immutable that it would complain and throw errors, but it's no longer necessary. You can pull the branch down and check for yourself. 👍

File: src/components/products/ProductCard.tsx [hunk 1]
```diff
@@ -1,26 +1,27 @@
-import { component$, mutable } from '@builder.io/qwik';
-import { Link } from '@builder.io/qwik-city';
+import { component$ } from '@builder.io/qwik';
 import Price from './Price';
 
 export default component$(
 	({ productAsset, productName, slug, priceWithTax, currencyCode }: any) => {
 		return (
-			<Link className="flex flex-col" href={`/products/${slug}`}>
+			<a className="flex flex-col" href={`/products/${slug}`}>
```
Comment (gioboa) [inline src/components/products/ProductCard.tsx:7]: I used <Link> because I prefer SPA behaviour, I don't like the whole refresh feeling

Comment (nnelgxorz) [reply src/components/products/ProductCard.tsx:7]: My mistake. I changed when I was getting the node_modules error on dev.ssr. I'll change it back in the morning and clean everything up.

Comment (gioboa) [reply src/components/products/ProductCard.tsx:7]: Great 👌

**Comment by gioboa** [comment]
I'm trying the application and I have mutable errors.if you click to "add to cart" button you will have this error

**Comment by michaelbromley** [comment]
Thanks @nnelgxorz for contributing to this project! I'm very happy to have a Qwik expert help out on this. I think we can turn this into a *very* compelling showcase for Qwik (and Vendure 😉)

**Comment by nnelgxorz** [comment]
Happy to contribute @michaelbromley. @gioboa, that is weird. I checked all the add to cart, remove from cart, change quantity and checkout behavior yesterday and it all worked. I'll dig in today and see what the issue is.

File: src/components/products/ProductCard.tsx (hunk 1 above)
Comment (nnelgxorz) [reply src/components/products/ProductCard.tsx]: Fixed!

**Comment by nnelgxorz** [comment]
Mutable props errors should be resolved now by moving by using AppState to avoid prop drilling issues. But again, double check and keep me honest. :+1:

**Pull request merged**